service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -139,75 +142,85 @@ else if (e1.getMessage().contains(Integer.toString(HttpStatus.EXPECTATION_FAILED
     private JsonObject erLookup(final JsonObject bodyJson, Boolean failIfMultipleFound, Boolean failIfNoneFound,
                final Boolean lookupInExternalERs, final int lookupLimit)
             throws REMGenerateException {
+        // Checking ER lookup limit
+        if(lookupLimit > 0) {
+            // Checking ER lookup enabled or not
+            if (erlookupConfig.getEventRepositoryEnabled() && bodyJson.toString().contains("%lookup%")) {
```

Comment:
First we should check if the er lookup is enabled or not then we can check the limit, without enabling the ER lookup what is the use in checking the limit ?